marshal/proto_test.go

```diff
@@ -0,0 +1,142 @@
+package marshal
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xitongsys/parquet-go/schema"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type subElem struct {
+	Val string
+}
+
+type testNestedElem struct {
+	SubElem     subElem
+	SubPtr      *subElem
+	SubList     []subElem
+	SubRepeated []*subElem
+	// Marshal ok but not write since it doesn't have corresponding primitive type
+	EmptyElem     struct{}
+	EmptyPtr      *struct{}
+	EmptyList     []struct{}
+	EmptyRepeated []*struct{}
+}
+type JobStatus int32
+
+const (
+	JobStatus_JobStatus_UNSPECIFIED  JobStatus = 0
+	JobStatus_BLOCKED                JobStatus = 1
+	JobStatus_ENQUEUED               JobStatus = 2
+	JobStatus_RUNNING                JobStatus = 3
+	JobStatus_COMPLETED              JobStatus = 4
+	JobStatus_ERRORED                JobStatus = 5
+	JobStatus_CANCELLED              JobStatus = 6
+	JobStatus_UPSTREAM_NOT_PROCESSED JobStatus = 7
+)
+
+func (x JobStatus) Enum() *JobStatus {
+	p := new(JobStatus)
+	*p = x
+	return p
+}
+
+func (x JobStatus) String() string {
+	statusToName := map[int32]string{
+		0: "ASYNCJOBSTATUS_UNSPECIFIED",
+		1: "BLOCKED",
+		2: "ENQUEUED",
+		3: "RUNNING",
+		4: "COMPLETED",
+		5: "ERRORED",
+		6: "CANCELLED",
+		7: "UPSTREAM_NOT_PROCESSED",
+	}
+	return statusToName[int32(x)]
+}
+
+type ProtoMessage struct {
+	Timestamp timestamppb.Timestamp
+	Status    JobStatus
+	IntVal    int32
+}
+
+func TestMarsalProtoSpecific(t *testing.T) {
+	protoMessages := []interface{}{
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 1, Nanos: 1000000}, Status: JobStatus_RUNNING, IntVal: 1},
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 2, Nanos: 1000000}, Status: JobStatus_ENQUEUED, IntVal: 2},
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 3, Nanos: 1000000}, Status: JobStatus_COMPLETED, IntVal: 3},
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 4, Nanos: 1000000}, Status: JobStatus_ERRORED, IntVal: 4},
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 5, Nanos: 1000000}, Status: JobStatus_CANCELLED, IntVal: 5},
+		ProtoMessage{Timestamp: timestamppb.Timestamp{Seconds: 6, Nanos: 1000000}, Status: JobStatus_UPSTREAM_NOT_PROCESSED, IntVal: 6},
+	}
+
+	schemaHandler, err := schema.NewSchemaHandlerFromProtoStruct(ProtoMessage{})
+	if err != nil {
+		t.Errorf("failed to get schema handler: %v", err)
+	}
+	tableMap, err := MarshalProto(protoMessages, schemaHandler)
+	if err != nil {
+		t.Errorf("failed to marshal values: %v", err)
+	}
+	assert.Equal(t, 3, len(schemaHandler.ValueColumns))
+	assert.Equal(t, 6, len((*tableMap)["Parquet_go_root\x01Timestamp"].Values))
```

Comment:
NIT: Please use a constant to avoid repeating this literal string `Parquet_go_root\x01Timestamp` several times.